### PR TITLE
feat(feature-page): dash0 comparison page

### PR DESCRIPTION
**Pull request by manika-signoz**
## Pull Request

---

Adds a new **SigNoz Cloud vs Dash0** comparison page at `/dash0-alternative`,.

#### Screenshots / Screen Recordings (if applicable)

| Hero + TL;DR | Why engineers choose SigNoz Cloud |
|---|---|
| 01-hero-tldr | 02-why-engineers-choose |

| Customer quotes | Comparison table |
|---|---|
| 03-quotes | 04-comparison-table |

| Distributed Tracing | Usage-based pricing calculator |
|---|---|
| 05-distributed-tracing-screenshots | 06-pricing-calculator |

| Verdict + inline CTA | Bottom CTA banner |
|---|---|
| 07-verdict-final-cta | 08-bottom-banner |

**Mobile (390px)**

| Hero | Table (horizontal scroll, sticky first column) |
|---|---|
| 09-mobile-hero | 10-mobile-table |

#### Issues closed by this PR

Closes SigNoz/growth-pod#1349

Addresses the VQA feedback in SigNoz/engineering-pod#6040 (homepage-dependent items deferred until #4120 merges).

---

### ✅ Change Type
_Select all that apply_

- [x] ✨ Feature
- [ ] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

- Tests added/updated: not required (marketing page built from existing shared components; no logic changes with test coverage expectations)
- Manual verification: Done on preview 
- Edge cases covered: mobile horizontal table scroll with sticky (translucent) first column; no page-level horizontal overflow at 390px; images lazy-load below the fold

---

- Blast radius: new route + registration touchpoints (TopNav comparison dropdown, footer "More" links, `ALTERNATIVES_SITEMAP_ROUTES`, `/product-comparison` index, `getPageType`). Shared-component change is limited to `ComparisonTable`, opt-in via `variant="striped"`.
- Potential regressions: `ComparisonTable` default rendering on `cloudwatch-alternative` / `llm-observability` / `observability-for-ai-native-companies` (manually verified unchanged).
- Rollback plan: revert the PR; the route and nav/footer/sitemap entries are self-contained

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested
- [x] Breaking changes documented (none)
- [x] Backward compatibility considered (`ComparisonTable` default variant unchanged)

---

## 👀 Notes for Reviewers

---

### CODE REVIEW

File: app/(site)/dash0-alternative/Dash0AlternativePage.constants.tsx [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+
+export const DASH0_HERO_BUTTONS: ButtonGroupButton[] = [
+  {
+    text: 'Get Started - Free',
+    href: '/teams/',
+    variant: 'default' as const,
+    className: BUTTON_CLASS_NAME,
+    tracking: {
+      clickType: 'Primary CTA',
+      clickName: 'Dash0 Alternative Hero Get Started',
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply app/(site)/dash0-alternative/Dash0AlternativePage.constants.tsx]: fixed

File: app/(site)/dash0-alternative/Dash0AlternativePage.tsx [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+  DASH0_HERO_BUTTONS,
+  DASH0_QUOTES,
+  DASH0_REASONS,
+  DASH0_TLDR,
+  DASH0_VENDORS,
+  DASH0_VERDICT,
+  DASH0_VERDICT_BUTTONS,
+  Dash0ComparisonBlock,
+} from './Dash0AlternativePage.constants'
+import UsageBasedPricing from 'shared/components/molecules/FeaturePages/UsageBasedPricing'
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply app/(site)/dash0-alternative/Dash0AlternativePage.tsx]: fixed

File: shared/components/molecules/FeaturePages/ComparisonTable/ComparisonTable.view.tsx [hunk 3]
```diff
@@ -33,29 +71,58 @@ const ComparisonTable: React.FC<ComparisonTableProps<string>> = ({
             ))}
           </tr>
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply shared/components/molecules/FeaturePages/ComparisonTable/ComparisonTable.view.tsx]: done, used thead

File: shared/components/molecules/FeaturePages/ComparisonTable/ComparisonTable.view.tsx [hunk 4]
```diff
@@ -1,30 +1,68 @@
...
+        ) : cell.supported === true ? (
+          <Check size={18} className="text-[var(--primary-background)]" />
+        ) : cell.supported === false ? (
+          <X size={18} className="text-red-400" />
+        ) : null}
+      </span>
+      <span>{cell.text}</span>
+    </span>
+  )
+}
```
[the remark this answers is not in the document]
Comment (manika-signoz) [reply shared/components/molecules/FeaturePages/ComparisonTable/ComparisonTable.view.tsx:33]: done, extracted a SupportIcon helper taking supported, size, and checkClassName

File: app/(site)/dash0-alternative/Dash0AlternativePage.tsx [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+      </h2>
+      <div className="mt-10">
+        {DASH0_REASONS.map((reason, index) => (
+          <div
+            key={reason.title}
+            className={`grid grid-cols-[44px_1fr] gap-4 border-t border-[var(--l1-border)] py-7 ${
+              index === DASH0_REASONS.length - 1 ? 'border-b' : ''
+            }`}
+          >
+            <div className="pt-1 font-mono text-sm text-[var(--l2-foreground)]">
```
Comment (H4ad) [inline app/(site)/dash0-alternative/Dash0AlternativePage.tsx]: You can mark this 01 numbers to be ignored by markdown

**Review by H4ad — COMMENTED** [review]
Do we need two CTA on the bottom? 

Image

File: app/(site)/dash0-alternative/Dash0AlternativePage.tsx (hunk 5 above)
Comment (manika-signoz) [reply app/(site)/dash0-alternative/Dash0AlternativePage.tsx]: done

**Review by H4ad — APPROVED** [review]
